packages/core/test/actorLogic.test.ts

```diff
@@ -982,4 +983,45 @@ describe('composable actor logic', () => {
       }
     });
   });
+
+  it('higher-level logic should work', () => {
+    const logged: any[] = [];
+    function withLogging<T extends ActorLogic<any, any>>(actorLogic: T) {
+      const enhancedLogic: T = {
+        ...actorLogic,
+        transition: (state, event, actorCtx) => {
+          logged.push(event.type);
+          return actorLogic.transition(state, event, actorCtx);
+        }
+      };
+
+      return enhancedLogic;
+    }
+
+    const machine = createMachine({
+      initial: 'start',
+      states: {
+        start: {
+          on: { next: 'working' }
+        },
+        working: {
+          on: { more: 'done' }
+        },
+        done: {}
+      }
+    });
+
+    const actor = createActor(withLogging(machine)).start();
+
+    actor.send({ type: 'next' });
+    actor.send({ type: 'more' });
+
+    expect(logged).toEqual(['next', 'more']);
+
+    expect(actor.getSnapshot().value).toBe('done');
+
+    expect(() => {
+      actor.getPersistedSnapshot();
+    }).not.toThrow();
```

Comment:
it might be worth testing the output of `getPersistedSnapshot` here. If it would return `undefined` or smth else (for whatever reason) then this test would still pass